cpu/esp_common/esp-now/esp_now_netdev.c

```diff
@@ -452,6 +453,31 @@ esp_now_netdev_t *netdev_esp_now_setup(void)
     return dev;
 }
 
+int esp_now_set_channel(uint8_t channel)
+{
+#ifdef ESP_NOW_UNICAST
+    scan_cfg.channel = channel;
+#endif
+#ifdef MODULE_ESP_WIFI
+    /* channel is controlled by `esp_wifi`, only update SoftAP info */
+    wifi_config_ap.ap.channel = channel;
+    return ESP_ERR_NOT_SUPPORTED;
```

Comment:
Should this also be returned when `ESP_NOW_UNICAST` is defined?
And why update SoftAP, is it ever used?

Reply:
> Should this also be returned when `ESP_NOW_UNICAST` is defined?

Why? `ESP_NOW_UNICAST` is just another mode where the scan channel has to be changed additionally. Whether the channel for the SoftAP interface can be changed or not only depends on whether `esp_wifi` is used.

> And why update SoftAP, is it ever used?

It is used if `esp_wifi` is not used.

Reply:
Alright